frontend/src/templates/Field/ChildrenCompound/ChildrenCompoundField.tsx

```diff
@@ -0,0 +1,404 @@
+  MyInfoAttribute,
+  MyInfoChildAttributes,
+  MyInfoChildData,
+  MyInfoChildVaxxStatus,
+} from '~shared/types'
+
+import { Button } from '~components/Button/Button'
+import { DatePicker } from '~components/DatePicker'
+import { SingleSelect } from '~components/Dropdown/SingleSelect'
+import { FormLabel } from '~components/FormControl/FormLabel/FormLabel'
+import { IconButton } from '~components/IconButton/IconButton'
+
+import { BaseFieldProps, FieldContainer } from '../FieldContainer'
+import {
+  ChildrenCompoundFieldInputs,
+  ChildrenCompoundFieldSchema,
+} from '../types'
+
+export interface ChildrenCompoundFieldProps extends BaseFieldProps {
+  schema: ChildrenCompoundFieldSchema
+  myInfoChildrenBirthRecords?: MyInfoChildData
+}
+
+/**
+ * Compound field for child information.
+ * This is "compound" because it can contain multiple subfields.
+ * The internal data representation is an array of arrays, where each
+ * subarray contains strings that represent the subfield array inputs.
+ *
+ * @precondition Must have a parent `react-hook-form#FormProvider` component.
+ */
+export const ChildrenCompoundField = ({
+  schema,
+  colorTheme = FormColorTheme.Blue,
+  myInfoChildrenBirthRecords,
+  ...fieldContainerProps
+}: ChildrenCompoundFieldProps): JSX.Element => {
+  const childrenInputName = useMemo(
+    () => `${schema._id}` as const,
+    [schema._id],
+  )
+
+  const formContext = useFormContext<ChildrenCompoundFieldInputs>()
+  const { isSubmitting } = useFormState<ChildrenCompoundFieldInputs>({
+    name: schema._id,
+  })
+
+  const { fields, append, remove } = useFieldArray<ChildrenCompoundFieldInputs>(
+    {
+      control: formContext.control,
+      name: `${schema._id}.child`,
+    },
+  )
+
+  useEffect(() => {
+    if (schema.childrenSubFields) {
+      formContext.setValue(
+        `${schema._id}.childFields`,
+        schema.childrenSubFields,
+      )
+    }
+  }, [schema.childrenSubFields, formContext, schema._id])
+
+  // Initialize with a single child section
+  useEffect(() => {
+    if (!fields || !fields.length || fields.length === 0) {
+      append([''])
+    }
+  }, [fields, append])
+
+  const ariaChildrenDescription = useMemo(() => {
+    let description = simplur`This is a children field. There [is|are] ${fields.length} child[|ren].`
+    description += `Each child has multiple fields to fill.`
+    description += `You can fill the child by selecting the child's name from the child name dropdown.`
+    if (schema.allowMultiple) {
+      description += ` You can add another child if you'd like by clicking the "Add another child" button below`
+    }
+
+    return description
+  }, [fields.length, schema.allowMultiple])
+
+  const numChild = fields.length ?? 0
+
+  return (
+    <FieldContainer
+      schema={schema}
+      {...fieldContainerProps}
+      errorKey={childrenInputName}
+    >
+      <VisuallyHidden id={`children-desc-${schema._id}`}>
+        {ariaChildrenDescription}
+      </VisuallyHidden>
```

Comment:
Good stuff! which reminds me we should test all this for accessibility too. @foochifa should be able to help